beginners-guide.md

```diff
@@ -19,10 +19,18 @@ SIMD has a few special vocabulary terms you should know:
 
 * **Vector:** A SIMD value is called a vector. This shouldn't be confused with the `Vec<T>` type. A SIMD vector has a fixed size, known at compile time. All of the elements within the vector are of the same type. This makes vectors *similar to* arrays. One difference is that a vector is generally aligned to its *entire* size (eg: 16 bytes, 32 bytes, etc), not just the size of an individual element. Sometimes vector data is called "packed" data.
 
+* **Vectorize**: An operation that uses SIMD instructions to operate over a vector is often referred to as "vectorized".
+
+* **Autovectorization**: Also known as _implicit vectorization_. This is when a compiler can automatically recognize a situation where scalar instructions may be replaced with SIMD instructions, and use those instead. A compiler lacks the same knowledge about the purpose of the program and shape of the data as the programmer who is using it, and that kind of domain knowledge strongly affects when and where SIMD instructions are appropriate, so this compiler optimization can often fail to trigger, even when it seems obvious to the programmer.
+
+* **Scalar:** "Scalar" in mathematical contexts refers to values that have only one "dimension", like real numbers (which can be represented by a line), but in programming it can also be used to describe "scalar operations" that use strictly scalar values, like addition. This term is mostly used to differentiate between vectorized operations that use SIMD instructions versus scalar operations that don't.
```

Comment:
Some of this phrasing might be confusing since complex numbers are scalar. Might be better to just say a scalar is one of something, vs a vector which is many

Reply:
Complex numbers don't exist and can't hurt you, they're imaginary. (noted)